simapp/v2/simdv2/cmd/commands.go

```diff
@@ -220,3 +180,43 @@ func appExport[T transaction.Tx](
 
 	return simApp.ExportAppStateAndValidators(forZeroHeight, jailAllowedAddrs, modulesToExport)
 }
+
+var _ transaction.Codec[transaction.Tx] = &temporaryTxDecoder[transaction.Tx]{}
+
+type temporaryTxDecoder[T transaction.Tx] struct {
```

Comment:
how come this is needed?

Reply:
I've just moved it down. I think we should make it not needed indeed